ledger/stateproofverificationtracker.go

```diff
@@ -0,0 +1,279 @@
+		spt.dbQueries.lookupStateProofVerificationData.Close()
+	}
+}
+
+func (spt *stateProofVerificationTracker) LookupVerificationData(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	spt.stateProofVerificationMu.RLock()
+	defer spt.stateProofVerificationMu.RUnlock()
+
+	if len(spt.trackedCommitData) > 0 && stateProofLastAttestedRound >= spt.trackedCommitData[0].verificationData.TargetStateProofRound &&
+		stateProofLastAttestedRound <= spt.trackedCommitData[len(spt.trackedCommitData)-1].verificationData.TargetStateProofRound {
+		return spt.lookupDataInTrackedMemory(stateProofLastAttestedRound)
+	}
+
+	if len(spt.trackedCommitData) == 0 || stateProofLastAttestedRound < spt.trackedCommitData[0].verificationData.TargetStateProofRound {
+		return spt.lookupDataInDB(stateProofLastAttestedRound)
+	}
+
+	return &ledgercore.StateProofVerificationData{}, fmt.Errorf("requested data for round %d, greater than maximum data round %d: %w",
+		stateProofLastAttestedRound,
+		spt.trackedCommitData[len(spt.trackedCommitData)-1].verificationData.TargetStateProofRound,
+		errStateProofVerificationDataNotFound)
+}
+
+func (spt *stateProofVerificationTracker) lookupDataInTrackedMemory(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	for _, commitData := range spt.trackedCommitData {
+		if commitData.verificationData.TargetStateProofRound == stateProofLastAttestedRound {
+			verificationDataCopy := commitData.verificationData
+			return &verificationDataCopy, nil
+		}
+	}
+
+	return &ledgercore.StateProofVerificationData{}, fmt.Errorf("%w for round %d: memory lookup failed",
+		errStateProofVerificationDataNotFound, stateProofLastAttestedRound)
+}
+
+func (spt *stateProofVerificationTracker) lookupDataInDB(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	verificationData, err := spt.dbQueries.lookupData(stateProofLastAttestedRound)
+	if err != nil {
+		err = fmt.Errorf("%w for round %d: %s", errStateProofVerificationDataNotFound, stateProofLastAttestedRound, err)
+	}
+
+	return verificationData, err
+}
+
+func (spt *stateProofVerificationTracker) committedRoundToLatestCommitDataIndex(committedRound basics.Round) int {
+	latestCommittedDataIndex := -1
+
+	for index, data := range spt.trackedCommitData {
+		if data.confirmedRound <= committedRound {
+			latestCommittedDataIndex = index
+		} else {
+			break
+		}
+	}
+
+	return latestCommittedDataIndex
+}
+
+func (spt *stateProofVerificationTracker) committedRoundToLatestDeleteDataIndex(committedRound basics.Round) int {
+	latestCommittedDataIndex := -1
+
+	for index, data := range spt.trackedDeleteData {
+		if data.confirmedRound <= committedRound {
+			latestCommittedDataIndex = index
+		} else {
+			break
+		}
+	}
+
+	return latestCommittedDataIndex
+}
+
+func (spt *stateProofVerificationTracker) insertCommitData(blk *bookkeeping.Block) {
```

Comment:
the name is confusing, insert might be though as DBMS insert

Reply:
renamed insert to append